### ISSUE TITLE
W: /home/alex/apktool/Working/name/res/values/public.xml:1736: error: resource 'drawable/$avd_hide_password__0' has invalid entry name '$avd_hide_password__0'. Invalid character '$avd_hide_password__0'.

### ISSUE THREAD

**Issue by Zaotar** [iBotPeaches/Apktool#2119]
### Information
1. **Apktool Version (`apktool -version`)** - 2.4.0
2. **Operating System (Mac, Linux, Windows)** - Linux
3. **APK From? (Playstore, ROM, Other)** - https://github.com/mozilla-mobile/fenix/releases

### Stacktrace/Logcat
```
W: invalid resource directory name: /home/alex/apktool/Working/name/res navigation
then I use --use-aapt2 option (I found this solution here)
then I see this:
W: /home/alex/apktool/Working/name/res/values/public.xml:1736: error: resource 'drawable/$avd_hide_password__0' has invalid entry name '$avd_hide_password__0'. Invalid character '$avd_hide_password__0'.
I found here recommendation to remove $ symbol and after this:
apktool b --use-aapt2 name -o v1.apk && java -jar signapk.jar testkey.x509.pem testkey.pk8 v1.apk v1-signed.apk && rm v1.apk
I: Using Apktool 2.4.0
I: Checking whether sources has changed...
I: Checking whether resources has changed...
I: Building resources...
W: /home/alex/apktool/Working/name/res/values/public.xml:1736: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/avd_hide_password__0'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1737: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/avd_hide_password__1'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1738: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/avd_hide_password__2'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1739: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/avd_show_password__0'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1740: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/avd_show_password__1'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1741: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/avd_show_password__2'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1742: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/ic_fx_accounts_avatar__0'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1743: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/ic_fx_accounts_avatar__1'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1744: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/ic_fx_accounts_avatar__2'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1745: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/ic_fx_accounts_avatar__3'.
W: /home/alex/apktool/Working/name/res/values/public.xml:2004: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/ic_launcher_foreground__0'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1746: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/ic_onboarding_avatar_anonymous__0'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1747: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/ic_onboarding_avatar_anonymous__1'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1748: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/ic_onboarding_firefox_accounts__0'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1749: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/ic_onboarding_firefox_accounts__1'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1750: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/ic_onboarding_privacy_notice__0'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1751: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/ic_onboarding_private_browsing__0'.
W: /home/alex/apktool/Working/name/res/values/public.xml:1752: error: no definition for declared symbol 'org.mozilla.fenix.nightly:drawable/ic_onboarding_tracking_protection__0'.
W: error: failed linking references.
```

### Steps to Reproduce
1. `apktool `

### Frameworks
If this APK is from an OEM ROM (Samsung, HTC, LG). Please attach framework files
(`.apks` that live in `/system/framework` or `/system/priv-app`)

### APK
If this APK can be freely shared, please upload/attach a link to it.

### Questions to ask before submission
1. Have you tried `apktool d`, `apktool b` without changing anything? yes
2. If you are trying to install a modified apk, did you resign it? yes
3. Are you using the latest apktool version? yes

**Comment by nanjingdaqi** [comment]
The resource `avd_hide_password.xml` is defined in support-design aar. It is a new kind of drawable `animated-vector`. 
https://developer.android.com/reference/android/graphics/drawable/AnimatedVectorDrawable

**Comment by iBotPeaches** [comment]
The fix for the filename is in v2.5.0 upcoming. The animated-vector is still a flat xml file, so should be handled without error.

**Comment by eladkarako** [comment]
still happens on 2.6.1 (nightly build). for some reason after turning `--verbose` ON, the build was a success. reproducible too! so weird.
